compiler/ActonCompiler.hs

```diff
@@ -290,6 +291,8 @@ findPaths actFile args  = do execDir <- takeDirectory <$> System.Environment.get
                                     return $ (False, False, pre, drop 1 ds)
 
 data CompileTask        = ActonTask  {name :: A.ModName, src :: String, atree:: A.Module} deriving (Show)
+-- TODO: why is this needed? why can't we use a A.GName?
+data RootActor          = RootActor {modname :: A.ModName, actname :: A.Name}
```

Comment:
@nordlander so this code works but it seems redundant to have to define a new record here. I want a record so that I easily can map over a list of such records and still pass all the necessary things.

I have a GName and later in buildExecutable we need a QName. Those records have the same field names, can I just convert / case one to the other?

I tried to pass GName, then extract the mname & noq and then build a QName, but it complains about trying to use mname. Is the mname field selector associated with QName? :confused:

Reply:
Discussed; replace record with one for BinTask or similar that also contains binFileName in addition to a A.GName

Reply:
Ok, done.